packages/block-editor/src/store/selectors.js

```diff
@@ -1036,6 +1036,9 @@ const canInsertBlockTypeUnmemoized = ( state, blockName, rootClientId = null ) =
 			return list;
 		}
 		if ( isArray( list ) ) {
+			if ( includes( list, 'root' ) && item === null ) {
```

Comment:
How would it work if it's not a post, in case of full site editing? Maybe not something we should fix now, but maybe good to have in mind. Cc @epiqueras.

Reply:
It would only let you insert it at the root level of the template.

I guess we would need to add `core/post-content` to the list of allowed parents? Along with any other template layout blocks that should be able to insert a page break.

Reply:
Could we also add a list of parents that should be considered as a 'root' container at this point, rather than having to add them to each blocks allowed parents? eg.

```javascript
const rootContainers = [ null, 'core/post-content ];
if ( includes( list, 'root' ) && includes( rootContainers , item )  )
```

Reply:
Sure, but won't the page break still break if its rendered in a Post Content block? Or does it only break in Column blocks?

Reply:
I just used  Post Content block as an example because you had mentioned it - the array would just contain any parent blocks that we knew should be treated like a root element for the likes of page breaks. 

It may be that it is not possible to identify what constitutes a 'root' block for a range of different child blocks, in which case it is probably best to leave 'root' purely as the editor root with `if ( includes( list, 'root' ) && item === null )`, and add any other possible parent blocks to the blocks own allowed parents array.

Reply:
>It may be that it is not possible to identify what constitutes a 'root' block for a range of different child blocks, in which case it is probably best to leave 'root' purely as the editor root with if ( includes( list, 'root' ) && item === null ), and add any other possible parent blocks to the blocks own allowed parents array.

A "root" block shouldn't be defined in relationship to a range of child blocks. A "root" block is simply a block that serves as the root of the block tree in some rendering context. For example, a Post Content block serves as the root of the block tree when rendering a single post, but doesn't when rendering an entire template. So I think we are safe with including Post Content in our definition of "root".

>I just used Post Content block as an example because you had mentioned it - the array would just contain any parent blocks that we knew should be treated like a root element for the likes of page breaks.

Yeah, I understood that. I was just wondering if the Page Break block will cause issues as a child of the Post Content block. I think it should be fine.

Now that I think more about this. The Page Break block only makes sense as a top level block in post content. If we whitelist this using "root", you would be able to add them to templates and template parts which wouldn't make any sense or work.

To make this work, the selector needs to check that the block is being inserted at the root level and that we are editing a post, or that it's being inserted into a Post Content block.

>that we are editing a post

We don't have a canonical way of doing this yet so I think that the way to move forward is using

```
parent: [ 'core/post-content' ]
```

and

```
if ( includes( list, 'core/post-content' ) && item === null ) {
```

for now. With a todo note to change it to

```
if (
  includes( list, 'core/post-content' ) &&
  (
    item === 'core/post-content' ||
    (getEditorMode() === 'post-content' && item === null)
  ) {
```

when possible.

Reply:
That approach makes sense to me - have made this change, but haven't had time to test it properly - will do that tomorrow hopefully.

Reply:
Looks good to me.

We'll actually only need to change it to `includes( list, 'core/post-content' ) && getEditorMode() === 'post-content' && item === null`, because the following `return includes( list, item );` will take care of the other case where `item === 'core/post-content'`.

Reply:
Have updated the TODO comment. Will try and get a unit test added in the next day or so.

Reply:
@epiqueras, finally got back to this, have added a basic unit test to cover this.